### ISSUE TITLE
Rocket frontend mask logic change of behavior

### ISSUE THREAD

**Issue by ccelio** [ucb-bar/rocket-chip#183]
Introduced by commit https://github.com/ucb-bar/rocket/commit/d91be2b545ea0eb704bf3c2b02b5d200f27426fa#diff-dbf1307484c9f6974955adde94a99967, the behavior of the masking logic has changed (https://github.com/ucb-bar/rocket-chip/blob/master/rocket/src/main/scala/frontend.scala#L128):

````scala
io.cpu.resp.bits.mask := UInt((1 << fetchWidth)-1) << s2_pc.extract(log2Up(fetchWidth)+log2Up(coreInstBytes)-1, log2Up(coreInstBytes))
````

For `fetchWidth=1`, `coreInstBytes=4`, that reduces to

> mask := UInt(1) << s2_pc(1+2-1, 2)
> mask := 1 << pc(2,2)

This means `mask` can be 0.

The old behavior, for `fetchWidth = 1`, was for `mask` to be set to UInt(1) (as the one instruction will always be valid if imem.valid is true).

Is the changed behavior intentional?

**Comment by aswaterman** [comment]
No; the problem is log2Up (it would have noisily failed using log2Ceil).
Will fix.

**Comment by ccelio** [comment]
While I have you, what's the meaning of "extract"? I assume this has different semantics from "()"; does it handle (-1,0)?

**Comment by aswaterman** [comment]
It's a widget I added to rocket utils, which returns UInt(0) when
attempting to extract (n-1, n).

**Comment by aswaterman** [comment]
This will be fixed by a6e009d8debf3a4a058275f8de9fc270e68ddfa2 once it gets merged
